### DISCUSSION TITLE
Add Canadian IDs

### DISCUSSION THREAD

**Discussion by andimik** [PE5PVB/TEF6686_ESP32#192] [Ideas]
Currently only USA is supported, but I found the Canadian conversion on

https://discord.com/channels/1053804249651359765/1053804250276298895/1123140592495501405

https://ised-isde.canada.ca/site/spectrum-management-telecommunications/en/licences-and-certificates/broadcasting/program-information-codes-radio-broadcasting-stations

![grafik](https://github.com/PE5PVB/TEF6686_ESP32/assets/24510556/338fe279-3f0f-4770-98df-4295d42e29e3)

Using USA will lead to useless ID conversion.

> 102.1 - K}GU > B203. Callsign CBR
103.1 - K[]WV - C37C. Callsign CFXL

Because the user from Canada has switched the country settings to USA in order to get 10 kHz grid on MW ...
https://discord.com/channels/1053804249651359765/1053804250276298895/1136970665627488277

**Comment by MCelliotG** [comment]
And here is the latest conversion formula, including Canada.
[rdsconverter.txt](https://github.com/PE5PVB/TEF6686_ESP32/files/12259945/rdsconverter.txt)

**Comment by andimik** [comment]
Is the source https://radio.feralfox.net/tools/rds.php ?

The author wrote:

>         /* Please seek permission before using the modified code - by sending a DM to Kyle on Discord - kaylon_sphere#9712 */

But I could not find him there. His last Youtube video is 8 months old. I don't have Instagram or Twitter.

Before implementing, we should get in contact with him first.

**Reply by MCelliotG** [reply]
No, the source is Casey Media. They are the ones who added the latest stuff.

**Reply by MCelliotG** [reply]
https://twitter.com/kaylon_sphere?lang=el

**Reply by MCelliotG** [reply]
https://twitter.com/dx_sphere

**Comment by PE5PVB** [comment]
I have no way of contacting him with X. Do you have other methodes of contacting him?

**Reply by HyperDX** [reply]
You can contact him via e-mail: dxsphere@gmail.com

**Comment by PE5PVB** [comment]
I'd like to invite programmers who know all about the callsign conversion in USA and Canada to write a C+ function which converts a 'uint16_t' PI-code to a 'char' callsign

**Comment by PE5PVB** [comment]
@andimik Do you know a website or do you have an excel document with all Canadian PI's, callsigns and frequencies?

**Reply by andimik** [reply]
No, sorry, I just found what I already have posted here.

**Comment by PE5PVB** [comment]
I'll close this one. Another topic is all about it.
